src/Microsoft.Data.SqlClient/tests/FunctionalTests/SqlConnectionBasicTests.cs

```diff
@@ -34,6 +35,33 @@ public void IntegratedAuthConnectionTest()
             connection.Open();
         }
 
+        [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.IsNotArmProcess))]
+        [InlineData(40613)]
+        [InlineData(42108)]
+        [InlineData(42109)]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public async Task TransientFaultTestAsync(uint errorCode)
+        {
+            using TransientFaultTDSServer server = TransientFaultTDSServer.StartTestServer(true, true, errorCode);
+            SqlConnectionStringBuilder builder = new()
+            {
+                DataSource = "localhost," + server.Port,
+                IntegratedSecurity = true,
+                Encrypt = SqlConnectionEncryptOption.Optional
+            };
+
+            using SqlConnection connection = new(builder.ConnectionString);
+            try
+            {
+                await connection.OpenAsync();
+                Assert.Equal(ConnectionState.Open, connection.State);
+            }
+            catch (Exception e)
+            {
+                Assert.False(true, e.Message);
+            }
```

Comment:
In the case of failure, the exception has better information.
```suggestion
            await connection.OpenAsync();
            Assert.Equal(ConnectionState.Open, connection.State);
```